internal/provisioner/utility/argo_utility.go

```diff
@@ -47,8 +47,15 @@ func ProvisionUtilityArgocd(utilityName, tempDir, clusterID string, allowCIDRRan
 		return errors.Wrap(err, "failed to write argo application file")
 	}
 
+	// Create the cluster output directory for the utility
+	clusterOutputDir := tempDir + "/apps/" + awsClient.GetCloudEnvironmentName() + "/helm-values/" + clusterID
```

Comment:
Just a heads-up that you can use `filepath.Join()` to do this sort of thing and reduce risk of building a bad path: https://pkg.go.dev/path/filepath#Join